plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sauditeks
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/alecthomas/jsonschema"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/symbols/extract"
+	"github.com/falcosecurity/plugins/plugins/k8saudit/pkg/k8saudit"
+	"github.com/falcosecurity/plugins/shared-aws/shared/go/aws/cloudwatchlogs"
+	"github.com/falcosecurity/plugins/shared-aws/shared/go/aws/session"
+)
+
+const pluginName = "k8sauditeks"
+
+type Plugin struct {
+	k8saudit.Plugin
+	Config PluginConfig
+}
+
+type PluginConfig struct {
+	Region          string `json:"region"           jsonschema:"description=The Region of your EKS cluster (default: us-east-1), it can be replaced by env var AWS_REGION"`
+	Profile         string `json:"profile"          jsonschema:"description=The Profile to use to create the session, it can be replaced by env var AWS_PROFILE"`
+	ClusterName     string `json:"cluster_name"     jsonschema:"description=The name of your EKS Cluster,required"`
+	UseAsync        bool   `json:"use_async"        jsonschema:"description=If true then async extraction optimization is enabled (Default: true)"`
+	PollingInterval uint64 `json:"polling_interval" jsonschema:"description=Polling Interval in seconds (default: 5s)"`
+	Shift           uint64 `json:"shift"            jsonschema:"description=Time shift in past in seconds (default: 1s)"`
+	BufferSize      uint64 `json:"buffer_size"      jsonschema:"description=Buffer Size (default: 200)"`
+}
+
+func (k *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          9,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for EKS from Cloudwatch Logs",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// setDefault sets the configuration to its default values
+func (k *PluginConfig) setDefault() {
+	if p := os.Getenv("AWS_DEFAULT_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if p := os.Getenv("AWS_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if r := os.Getenv("AWS_DEFAULT_REGION"); r != "" {
+		k.Region = r
+	}
+	if r := os.Getenv("AWS_REGION"); r != "" {
+		k.Region = r
+	}
+	if k.Region == "" {
+		k.Region = "us-east-1"
```

Comment:
I think region should not be optional here. Another point in favor of making it an open param.

Reply:
Ok

Reply:
For the region, I don't think we should make it required, I don't think either we should set a default value. Most users will run Falco on EKS, and by design, the SDK will access to the correct endpoints of AWS API. The region is only used for that. wydt?